modulo2/projeto-labefy/labefy/src/components/PLDetails.js

```diff
@@ -0,0 +1,113 @@
+import axios from 'axios'
+import React, { Component } from 'react'
+import { PLDetailContainer, PLDetailHeaderContainer, TrackContainer, PLDetailAddContainer, ContainerHeader, ContainerLogo, ContainerMenu } from '../styles'
+import logo_claro from '../img/musica_clara.png'
+
+export default class PLDetails extends Component {
+
+    state = {
+        playlistTracks: [],
+        playlistTracksQty: 0,
+        newMusicName: '',
+        newMusicArtist: '',
+        newMusicURL: ''
+    }
+
+    componentDidMount() {
+        this.getPlaylistDetails()
+    }
+
+    getPlaylistDetails = () => {
+        axios.get(`https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists/${this.props.id}/tracks`, {
+            headers: {
+                Authorization: 'cesar-huber-carver'
+            }
+        })
+            .then((res) => {
+                this.setState({ playlistTracks: res.data.result.tracks })
+                this.setState({ playlistTracksQty: res.data.result.quantity })
+            })
+            .catch((err) => {
+                alert(err.response.data.message)
+            })
+    }
+
+    handleNewMusicNameChange = (e) => {
+        this.setState({ newMusicName: e.target.value })
+    }
+
+    handleNewMusicArtistChange = (e) => {
+        this.setState({ newMusicArtist: e.target.value })
+    }
+
+    handleNewMusicURLChange = (e) => {
+        this.setState({ newMusicURL: e.target.value })
+    }
+
+    addTrackToPlaylist = () => {
+        const url = `https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists/${this.props.id}/tracks`
+        const body =
+        {
+            name: this.state.newMusicName,
+            artist: this.state.newMusicArtist,
+            url: this.state.newMusicURL
+        }
+        axios.post(url, body, { headers: { Authorization: 'cesar-huber-carver' } })
+            .then((res) => {
+                alert('Música adicionada com sucesso')
+                this.getPlaylistDetails()
+                this.setState({ newMusicName: '' })
+                this.setState({ newMusicArtist: '' })
+                this.setState({ newMusicURL: '' })
```

Comment:
Nesse caso a forma recomendada a se fazer com o this.setState é ultiliza-lo em uma única chamada:
```
this.setState({
  newMusicName: ',
  newMusicArtist: '',
  newMusicURL: ''

})
```